scipy/special/wright_bessel.pxd

```diff
@@ -0,0 +1,685 @@
+        res += exp(k * lnx - lgam(k + 1) - lgam(a*k + b))
+
+    return res
+
+
+# 3. Taylor series in a=0 up to order 5, for tiny a and not too large x
+# Phi(a, b, x) = exp(x)/Gamma(b)
+#               * (1 - a*x*Psi(b) + a^2/2*x*(1+x)(Psi(b)^2 - Psi'(b)) + O(a^3))
+# where Psi is the digamma function.
+# Call: python _precompute/wright_bessel.py 1
+#
+# For small b, cancellation of poles of digamma(b)/Gamma(b) and polygamma needs
+# to be carried out => series expansion in a and b around 0 to order 5.
+# Call: python _precompute/wright_bessel.py 2
+@cython.cdivision(True)
+cdef inline double _wb_small_a(double a, double b, double x, int order) nogil:
+    cdef:
+        double dg, dg1, dg2, dg3, res
+        double A[6]  # powers of a^k/k!
+        double B[5]  # powers of b^k/k!
+        double C[5]  # coefficients of a^k1 * b^k2
+        int k
+
+    if b <= 1e-3:
+        # Series expansion of both a and b up to order 5:
+        C[0] = 1
+        C[1] = 2*M_EG
+        C[2] = 3*M_EG**2 - M_PI**2/2
+        C[3] = 4*M_EG**3 - 2*M_EG*M_PI**2 + 8*M_Z3
+        C[4] = 5*M_EG**4 - 5*M_EG**2*M_PI**2 + 40*M_EG*M_Z3 + M_PI**4/12
+        A[0] = 1.
+        B[0] = 1.
+        for k in range(1, 5):
+            A[k] = a/k * A[k-1]
+            B[k] = b/k * B[k-1]
+        A[5] = a/5*A[4]
+        res = rgamma(b)
+        res += a*x * (C[0] + C[1]*b + C[2]*B[2] + C[3]*B[3] + C[4]*B[4])
+        res += A[2]*x*(1+x)*(C[1]   + C[2]*b    + C[3]*B[2] + C[4]*B[3])
+        res += A[3]*x*(x**2+3*x+1) * (C[2]      + C[3]*b    + C[4]*B[2])
+        res += A[4]*x*(x**3+6*x**2+7*x+1) *      (C[3]      + C[4]*b)
+        res += A[5]*x*(x**4+10*x**3+25*x**2+15*x+1) *         C[4]
+        res *= exp(x)
+    else:
+        dg = digamma(b)
+        # dg1 = polygamma(1, b)
+        dg1 = zeta(2, b)
+        # res = 1 - a*x*dg + a**2/2*x*(1+x)*(dg**2 - dg1)
+        res = 1 + a*x*(-dg + 0.5*a*(1 + x)*(dg**2 - dg1))
+        if order >= 3:
+            # dg2 = polygamma(2, b)
+            dg2 = -2. * zeta(3, b)
+            res += -a**3/6. * x*(x**2 + 3*x + 1)*(dg**3 - 3*dg*dg1 + dg2)
+        if order >= 4:
+            # dg3 = polygamma(3, b)
+            dg3 =  6 * zeta(4, b)
+            res += a**4/24. * x*(x**3 + 6*x**2 + 7*x + 1) \
+                   *(dg**4 - 6*dg**2*dg1 + 4*dg*dg2 + 3*dg1**2 - dg3)
+        res *= exp(x) * rgamma(b)
+    return res
+
+
+# 4. Asymptotic expansion for large x up to order 8
+#    Phi(a, b, x) ~ Z^(1/2-b) * exp((1+a)/a * Z) * sum_k (-1)^k * C_k / Z^k
+#    with Z = (a*x)^(1/(1+a)).
+#
+# Call: python _precompute/wright_bessel.py 3
+@cython.cdivision(True)
+cdef inline double _wb_asymptotic(double a, double b, double x) nogil:
+    cdef:
+        double A[15]  # powers of a
+        double B[17]  # powers of b
+        double Ap1[9]  # powers of (1+a)
+        double C[9]  # coefficients of asymptotic series a_k
+        double Z, Zp, res
+        int k
+
+    A[0] = 1.
+    B[0] = 1.
+    Ap1[0] = 1.
+    for k in range(1, 15):
```

Comment:
Given that we are building a power basis for `a`, it seems likely that parts of this function could be done with `polevl`, which is going to be both faster and more accurate-it's in `special/cephes/polevl.h`.

Reply:
IIUC, `polevl` uses the Horner form, which has mainly advantageous in accuracy. As in the current implementation, we can reuse the stored powers of a and b (in A and B), I think there is no significant performance gain. What we would need is a 2d or 3d polynomial evaluation function:smirk:
Given that the current version is more readable and it would be a larger refactoring, I'd prefer to keep it this way.